### PR TITLE
doc: update PyZMQ install instructions, fix zmq_sub.py file permissions

### PR DESCRIPTION
**Pull request by jonatack**
Seen while reviewing #19572.

### CODE REVIEW

File: doc/zmq.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Block and Transaction Broadcasting with ZeroMQ
 
-[ZeroMQ](http://zeromq.org/) is a lightweight wrapper around TCP
+[ZeroMQ](https://zeromq.org/) is a lightweight wrapper around TCP
```
Comment (laanwj) [inline doc/zmq.md:3]: How are we stil finding http links in the repository in the year 2020 :sweat_smile:

Comment (practicalswift) [reply doc/zmq.md:3]: Thanks for fixing @jonatack! :)

A `scripted-diff` might be needed to bring the remaining old school style links to year 2020 :)

```
$ git grep 'http://www.opensource.org/licenses/mit-license.php' | wc -l
1092
```

Bonus points if we can remove the external -- and thus potentially changing -- reference to simply referencing the `COPYING` file. That would get rid of the horrible `php` extension too :)

Comment (jonatack) [reply doc/zmq.md:3]: True :dango: :icecream:  though updating all those links isn't really in the scope of this patch. I actually only noticed the http url just before opening it. :shipit:

**Comment by jonatack** [comment]
If I'm not mistaken, `contrib/zmq/zmq_sub.py` had the wrong file permissions; changed them to `755`.

**Comment by jonatack** [comment]
If I'm not mistaken, `contrib/zmq/zmq_sub.py` had the wrong file permissions; changed them to `755`.

There are other scripts in `contrib` that seem to have the same issue. LMK if I should update those in the same commit.

File: doc/zmq.md (hunk 1 above)
Comment (laanwj) [reply doc/zmq.md:3]: Ahhhhh updating 1002 files sounds … pretty awful … that fully answers my question though. Together with XML schemas and such it makes it hard to search for relevant ones. 
Agree it's out of scope of this PR.

**Review by theStack — APPROVED** [review]
ACK 062e6699c4ac48c3d46516190ec411dec3680a0d 🧷 
Checked that the changed/introduced URLs are visitable and correct and that the PyZMQ install instruction via pip works (tested in Ubuntu Xenial). Also ran the zmq_sub.py script directly to double-check that the execution flag is set now.

**Review by fanquake — APPROVED** [review]
ACK 062e6699c4ac48c3d46516190ec411dec3680a0d

**Pull request merged**